MapboxNavigation/RouteMapViewController.swift

```diff
@@ -219,12 +219,15 @@ class RouteMapViewController: UIViewController, PulleyPrimaryContentControllerDe
                 controller.distanceLabel.isHidden = true
             }
 
+            let streetLabelWidth = controller.streetLabel.bounds.width
+            let streetLabelFont = controller.streetLabel.font!
+            
             if let name = upComingStep.names?.first {
-                controller.streetLabel.text = name
+                controller.streetLabel.text = name.stringByAbbreviatingToFitWidth(width: streetLabelWidth, font: streetLabelFont)
             } else if let destinations = upComingStep.destinations?.joined(separator: "\n") {
-                controller.streetLabel.text = destinations
+                controller.streetLabel.text = destinations.stringByAbbreviatingToFitWidth(width: streetLabelWidth, font: streetLabelFont)
             } else {
-                controller.streetLabel.text = upComingStep.instructions
+                controller.streetLabel.text = upComingStep.instructions.stringByAbbreviatingToFitWidth(width: streetLabelWidth, font: streetLabelFont)
```

Comment:
I just noticed that we aren’t using the localized instructions coming from OSRMTextInstructions here. 🙀